src/utils.jl

```diff
@@ -208,13 +208,7 @@ inverse_scale_func(scale::Symbol) =
 # ## Unzip
 # --------------------------------
 
-for i in 2:4
-    @eval begin
-        unzip(v::AVec{<:Tuple{Vararg{T, $i} where T}}) =
-            $(Expr(:tuple, (:([t[$j] for t in v]) for j in 1:i)...))
-    end
-end
-
+unzip(v::AVec{<:Tuple}) = tuple((([t[j] for t in v]) for j in 1:length(v[1]))...)
```

Comment:
This is not entirely equivalent, since it will also unzip Tuples no matter how large

Reply:
Should we have used:
```julia
for i in 2:4
    @eval begin
        unzip(v::AVec{<:NTuple{$i,T} where T}) = $(Expr(:tuple, (:([t[$j] for t in v]) for j in 1:i)...))
    end
end
```
instead ?

Reply:
Potentially, would be interesting to benchmark the two for long vectors of tuples, but in the end I don't think it matters much